rust/src/cpu.rs

```diff
@@ -0,0 +1,1061 @@
+                // SED
+                (Instruction::Sed, _) => self.sed(),
+                // SEI
+                (Instruction::Sei, _) => self.sei(),
+                // CLC
+                (Instruction::Clc, _) => self.clc(),
+                // CLD
+                (Instruction::Cld, _) => self.cld(),
+                // CLI
+                (Instruction::Cli, _) => self.cli(),
+                // CLV
+                (Instruction::Clv, _) => self.clv(),
+                // BMI
+                (Instruction::Bmi, _) => self.bmi(&opcode),
+                // BPL
+                (Instruction::Bpl, _) => self.bpl(&opcode),
+                // BVS
+                (Instruction::Bvs, _) => self.bvs(&opcode),
+                // BVC
+                (Instruction::Bvc, _) => self.bvc(&opcode),
+                // BCS
+                (Instruction::Bcs, _) => self.bcs(&opcode),
+                // BCC
+                (Instruction::Bcc, _) => self.bcc(&opcode),
+                // BEQ
+                (Instruction::Beq, _) => self.beq(&opcode),
+                // BNE
+                (Instruction::Bne, _) => self.bne(&opcode),
+                // Other
+                _ => todo!("Code: {:x?} not implemented!", code),
+            };
+
+            self.update_pc(current_pc, opcode.bytes);
+        }
+    }
+
+    fn update_pc(&mut self, current_pc: u16, bytes: u8) {
+        if current_pc == self.cpu.program_counter {
+            self.cpu.program_counter += (bytes - 1) as u16;
+        }
+    }
+
+    //Operations for transferring bytes of data
+    fn lda(&mut self, opcode: &OpCode) {
+        let address = self.get_operand_address(&opcode.address_mode);
+        let value = self.mem_read_8(address);
+
+        self.cpu.accumulator = value;
+        self.cpu.update_zero_and_negative_flags(value);
+    }
+
+    fn ldx(&mut self, opcode: &OpCode) {
+        let adderss = self.get_operand_address(&opcode.address_mode);
+        let value = self.mem_read_8(adderss);
+
+        self.cpu.register_x = value;
+        self.cpu.update_zero_and_negative_flags(value);
+    }
+
+    fn ldy(&mut self, opcode: &OpCode) {
+        let address = self.get_operand_address(&opcode.address_mode);
+        let value = self.mem_read_8(address);
+
+        self.cpu.register_y = value;
+        self.cpu.update_zero_and_negative_flags(value);
+    }
+
+    fn sta(&mut self, opcode: &OpCode) {
+        let address = self.get_operand_address(&opcode.address_mode);
+
+        self.mem_write_8(address, self.cpu.accumulator);
+    }
+
+    fn stx(&mut self, opcode: &OpCode) {
+        let address = self.get_operand_address(&opcode.address_mode);
+
+        self.mem_write_8(address, self.cpu.register_x)
+    }
+
+    fn sty(&mut self, opcode: &OpCode) {
+        let address = self.get_operand_address(&opcode.address_mode);
+
+        self.mem_write_8(address, self.cpu.register_y)
+    }
+
+    fn tax(&mut self, opcode: &OpCode) {
```

Comment:
Is `OpCode` necessary for this instruction?

Reply:
No, I don't need it. I overlooked this because of the use of `#[allow(unused)]`. It was a temporary solution, but now I have removed it.